apps/reader/workers/notification-worker/plugins/publications.ts

```diff
@@ -0,0 +1,54 @@
+import {capture_event} from "posthog";
+import schema from "../../../schema";
+import {EVENT_NAMES, ProcessorPlugin} from "../../replicate-worker/helpers";
+import {NotificationProcessorPlugin, sendNotificationToPostHog} from "../helpers";
```

Comment:
When we need functionality, like lets say sending a notification, or monitoring an event's success or failure. 
We should pass it down as a parameter to the processor function(within an instance od a class, like the monitor class). 
e.g for the other processors, we pass in an instance of the monitor. 

For consistency do that here as well(I'm reffering to the sendNotificationToPostHog function)